salt/channel/server.py

```diff
@@ -33,48 +41,28 @@
 log = logging.getLogger(__name__)
 
 
-# TODO: rename
-class AESPubClientMixin:
-    def _verify_master_signature(self, payload):
-        if self.opts.get("sign_pub_messages"):
-            if not payload.get("sig", False):
-                raise salt.crypt.AuthenticationError(
-                    "Message signing is enabled but the payload has no signature."
-                )
-
-            # Verify that the signature is valid
-            master_pubkey_path = os.path.join(self.opts["pki_dir"], "minion_master.pub")
-            if not salt.crypt.verify_signature(
-                master_pubkey_path, payload["load"], payload.get("sig")
-            ):
-                raise salt.crypt.AuthenticationError(
-                    "Message signature failed to validate."
-                )
-
-    @salt.ext.tornado.gen.coroutine
-    def _decode_payload(self, payload):
-        # we need to decrypt it
-        log.trace("Decoding payload: %s", payload)
-        if payload["enc"] == "aes":
-            self._verify_master_signature(payload)
-            try:
-                payload["load"] = self.auth.crypticle.loads(payload["load"])
-            except salt.crypt.AuthenticationError:
-                yield self.auth.authenticate()
-                payload["load"] = self.auth.crypticle.loads(payload["load"])
-
-        raise salt.ext.tornado.gen.Return(payload)
-
-
-# TODO: rename?
-class AESReqServerMixin:
+class ReqServerChannel:
     """
-    Mixin to house all of the master-side auth crypto
+    ReqServerChannel handles request/reply messages from ReqChannels. The
+    server listens on the master's ret_port (default: 4506) option.
     """
 
-    def pre_fork(self, _):
+    @classmethod
+    def factory(cls, opts, **kwargs):
+        if "master_uri" not in opts and "master_uri" in kwargs:
+            opts["master_uri"] = kwargs["master_uri"]
```

Comment:
**question** would setdefault read better here? i.e. `opts.setdefault('master_uri', kwargs.get('master_uri'))`? It's not *technically* equivalent, as this is also hinging on `'master_uri' in kwargs` vs `kwargs.get`. Presumably there's handling elsewhere for a missing `master_uri` in general?

Reply:
Yes, this is another thing that should get cleaned up with the addition of a "connection details" class. The work to make that happen will span all the way from config to channels where we instantiate transports.